File: backend/app/schemas/ranking.py
from pydantic import BaseModel


class RankingEntry(BaseModel):
    rank: int
    character_name: str
    user_nickname: str | None
    rarity_score: float
    worldline_count: int
    race: str


class RankingResponse(BaseModel):
    rankings: list[RankingEntry]
    total_characters: int
    my_rank: int | None = None

File: ios/AnotherWorldline/Models/Ranking.swift
import Foundation

struct RankingEntry: Codable, Identifiable {
    var id: Int { rank }
    let rank: Int
    let characterName: String
    let userNickname: String?
    let rarityScore: Double
    let worldlineCount: Int
    let race: String

    enum CodingKeys: String, CodingKey {
        case rank, race
        case characterName = "character_name"
        case userNickname = "user_nickname"
        case rarityScore = "rarity_score"
        case worldlineCount = "worldline_count"
    }
}

struct RankingResponse: Codable {
    let rankings: [RankingEntry]
    let totalCharacters: Int
    let myRank: Int?

    enum CodingKeys: String, CodingKey {
        case rankings
        case totalCharacters = "total_characters"
        case myRank = "my_rank"
    }
}
